source/slang/ir.cpp

```diff
@@ -4223,7 +4218,7 @@ namespace Slang
                 continue;
 
             auto decoration = (IRTargetIntrinsicDecoration*) dd;
-            if(decoration->targetName == targetName)
+            if(String(decoration->targetName) == targetName)
```

Comment:
After seeing a bunch of these edit sites, I am now realizing that we should be adding an accessor (`decoration->getTargetName()`) rather than pushing this choice of representation down to all the use sites.

I'm certainly not one of those people who insists that *everything* should have an accessor, but my usual rule of thumb is that if I'm changing the way a property gets stored and I'm about to go and modify all the use sites, I go ahead and wrap things in accessor methods under the assumption that once I've changed the representation once, the probability of having to change it again goes up, so that the cost of using accessors is more likely to pay off. Just my opinion, though.